src/main/java/sx/blah/discord/util/cache/CacheMap.java

```diff
@@ -0,0 +1,54 @@
+/*
+ *     This file is part of Discord4J.
+ *
+ *     Discord4J is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU Lesser General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     Discord4J is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU Lesser General Public License for more details.
+ *
+ *     You should have received a copy of the GNU Lesser General Public License
+ *     along with Discord4J.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package sx.blah.discord.util.cache;
+
+import com.koloboke.collect.set.LongSet;
+import com.koloboke.compile.KolobokeMap;
+
+import java.util.Collection;
+
+@KolobokeMap
```

Comment:
For all methods which return `Map<Long, Obj>`, make it return the specialized map instead. As auto-boxing makes it more inefficient to use the generic Map insterface